### PR TITLE
Added support for caching solc-js different versions

### PR DESCRIPTION
**Pull request by eswarasai**
### Description
- [x] Added cache for solc-js versions

Fixes: #25

### CODE REVIEW

**Comment by eswarasai** [comment]
@wuestholz -- Resolved [this](https://github.com/b-mueller/sabre/pull/29#issuecomment-477577033) [here](https://github.com/b-mueller/sabre/pull/31/commits/2cb3f96b555dc83ecbe05498c491a30058e12b1b)

cc: @b-mueller

**Comment by Skyge** [comment]
OK!

File: lib/helpers.js [hunk 1]
```diff
@@ -192,14 +195,14 @@ const getSolidityVersion = fileContents => {
...
-                    upperLimit = solidityVersion.substring(solidityVersion.length-5, solidityVersion.length);
+                    upperLimit = solidityVersion.substring(solidityVersion.length - 5, solidityVersion.length);
                 }
-            }else if (solidityVersion.indexOf('>') !== -1) {
+            } else if (solidityVersion.indexOf('>') !== -1) {
                 solidityVersion = releases.latest;
             } else {
-                upperLimit = '0.' + (parseInt(solidityVersion[2])+1).toString() + '.0'
+                upperLimit = '0.' + (parseInt(solidityVersion[2]) + 1).toString() + '.0';
             }
```
Comment (Skyge) [inline lib/helpers.js]: @eswarasai Thanks for your correcting my spelling mistakes. I think I should extend a plugin for js in my vs code.

Comment (eswarasai) [reply lib/helpers.js]: Sure. No problem.

Yeah, about linting, it's always better to have good IDE support which will ease up development. Also, I'll be setting up precommit hooks to fix the lint issues before committing the code for the ease of maintenance and code review :)

**Pull request merged**